packages/ui-react/src/components/Image/Image.tsx

```diff
@@ -16,14 +15,46 @@ const setWidth = (url: string, width: number) => {
   return createURL(url, { width });
 };
 
+const cache = new Map();
+
+const resolveImageURL = async (imgUrl: string, width: number) => {
+  let url = setWidth(imgUrl, width);
+
+  if (cache.has(url)) {
+    return cache.get(url);
+  }
+
+  const response = await fetch(url);
+
+  // if redirected, cache and return resolved URL
+  if (response.redirected) {
+    url = response.url.replace('-1920', `-${width}`);
+  }
+
+  cache.set(url, url);
+
+  return url;
+};
```

Comment:
The `cache.set(url, url);` is still wrong because it uses the changed url as key. This causes the cache to never be hit.

```suggestion
const resolveImageURL = async (imgUrl: string, width: number) => {
  const requestUrl = setWidth(imgUrl, width);
  let url = requestUrl;

  if (cache.has(requestUrl)) {
    return cache.get(requestUrl);
  }

  const response = await fetch(requestUrl);

  // if redirected, cache and return resolved URL
  if (response.redirected) {
    url = response.url.replace('-1920', `-${width}`);
  }

  cache.set(requestUrl, url);

  return url;
};
```

Reply:
Apologees!